org.eclipse.dartboard/src/org/eclipse/dartboard/pub/PubService.java

```diff
@@ -0,0 +1,163 @@
+/*******************************************************************************
+ * Copyright (c) 2019 vogella GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Jonas Hungershausen - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.dartboard.pub;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Singleton;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.SubMonitor;
+import org.eclipse.core.runtime.jobs.IJobChangeEvent;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.core.runtime.jobs.JobChangeAdapter;
+import org.eclipse.dartboard.Messages;
+import org.eclipse.dartboard.util.StatusUtil;
+import org.eclipse.osgi.util.NLS;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link Singleton} service that exposes various Pub functions.
+ * 
+ * Currently the main purpose of this class is to execute the {@code pub get} in
+ * a given directory (project).
+ * 
+ * @author Jonas Hungershausen
+ * @see PubspecChangeListener
+ * 
+ */
+public class PubService {
+
+	private static final Logger LOG = LoggerFactory.getLogger(PubService.class);
+
+	private static PubService instance;
+
+	private PubService() {
+	}
+
+	/**
+	 * A {@link Map} holding an {@link IProject} and a {@link Job} as ideally there
+	 * should only be one {@code pub get} job running per project.
+	 */
+	private Map<IProject, Job> pubGetJobs = new HashMap<>();
```

Comment:
I am not sure about the flow of this class. Can you check do we need to clear or making null.

Reply:
> I am not sure about the flow of this class. Can you check do we need to clear or making null.

I'm not sure what you mean by this. The Map should always be the same object for the lifetime of the object.